gwsumm/archive.py

```diff
@@ -178,74 +178,86 @@ def read_data_archive(sourcefile):
     ----------
     sourcefile : `str`
         path to source HDF5 file
+
+    rm_source_on_fail : `bool`, optional
+        remove the source HDF5 file if there was an OSError when opening the
+        file
     """
     from h5py import File
 
-    with File(sourcefile, 'r') as h5file:
+    try:
+        with File(sourcefile, 'r') as h5file:
```

Comment:
Just an observation: In tracking down the corrupted files Robert found that some h5 utilities like h5ls appear to work while h5dump (and summary pages) fail. I am not sure but I suspect many corrupted files will get past the open statement.

Reply:
@duncanmmacleod Ok, sounds good. I was attempting to keep it symmetric with the `write_data_archive()` function, but this is somewhat cleaner with your suggestion. I have made your suggestion now.

@areeda True, this may or may not solve all our problems, but I'm hopeful it will solve a major issue that has been plaguing us for a long time. Time will tell...